rand_core/src/block.rs

```diff
@@ -28,8 +28,7 @@
 //! struct MyRngCore;
 //!
 //! impl BlockRngCore for MyRngCore {
-//!     type Item = u32;
-//!     type Results = [u32; 16];
+//!     type Results = [u8; 32];
```

Comment:
Note that right now `Default` is not implemented for arrays larger than 32. It unfortunately means that ChaCha20 implementation would have to use a newtype wrapper around array. We could use `mem::zeroed()`, but in general it would be unsafe, since in theory `Results` can be a struct containing other field in addition to array and its observed via `generate` function.

Reply:
Alternatively, we could define our own trait for initializing arrays. However, this would unfortunately require a breaking change if we want to switch back to `Default` once const generics are stabilized.

However, we are using a newtype in our ChaCha implementation anyway (already manually implementing `Default`), so I don't really see an issue with this.